src/audio.rs

```diff
@@ -478,12 +496,17 @@ impl Audio {
 struct Input {
     index: usize,
     buffer: &'static DmaBuffer,
+    max_transfer_size: usize,
 }
 
 impl Input {
     /// Create a new Input from a DmaBuffer
-    fn new(buffer: &'static DmaBuffer) -> Self {
-        Self { index: 0, buffer }
+    fn new(buffer: &'static DmaBuffer, max_transfer_size: usize) -> Self {
+        Self {
+            index: 0,
+            buffer,
+            max_transfer_size,
```

Comment:
It sucks that this needs to be bloated with another variable, but I don't see a way around it, since it needs to be dynamic and checked in a few places.

Reply:
This looks like a great candidate for easy compiler optimization!